multicluster/controllers/multicluster/member/clusterset_controller.go

```diff
@@ -133,6 +140,14 @@ func (r *MemberClusterSetReconciler) Reconcile(ctx context.Context, req ctrl.Req
 				return ctrl.Result{}, err
 			}
 		}
+		// When the installedLeader is not empty, it means a ClusterSet was initialized at least once,
+		// and we should try to clean up all stale resources before a new CommonArea is created.
+		if r.installedLeader != (leaderClusterInfo{}) {
```

Comment:
This is too implicit. Consider adding another flag to `MemberClusterSetReconciler`, like `cleanupNeeded`